src/sysutil.rs

```diff
@@ -44,10 +45,20 @@ pub struct SysUtil {
 impl SysUtil {
     pub fn new() -> IOResult<Self> {
         let (users, shells) = Self::init_users()?;
+        let mut sessions = Self::init_sessions(
+            XSESSION_DIRS,
+            "xsessions",
+            &vec!["startx".into(), "/bin/env".into()],
+        )?;
+        sessions.extend(Self::init_sessions(
+            WAYLAND_SESSION_DIRS,
+            "wayland-sessions",
+            &Vec::new(),
+        )?);
```

Comment:
Something to look at would be adding support for running e.g. Cage and starting Xwayland rootful to use X sessions :)